File: CSharp part2/03. Methods/05.LargerThanNeighbours/LargerThanNeighbours.cs
// Write a method that checks if the element at given position in given array of integers is larger than its two neighbours (when such exist).

using System;

namespace LargerThanNeighbours
{
    public class BiggerThanNeighbours
    {
        public static bool CompareNeighbours(int[] arr, int position)
        {
            bool isLarger = false;
            if (position < arr.Length && position > 0)
            {
                if (arr[position] > arr[position - 1] && arr[position] > arr[position + 1])
                {
                    isLarger = true;
                }
            }
            return isLarger;
        }
        static void Main()
        {
            Console.WriteLine("We are given array with elements");
            int[] array = { 5, 13, 34, 20, 5, 10, 45, 30 };
            foreach (int number in array)
            {
                Console.Write(number + ", ");
            }
            Console.Write("\nEnter position:");
            int position = int.Parse(Console.ReadLine());
            if (position == array.Length - 1 || position == 0)
            {
                Console.WriteLine("Element at position {0} doesn't have two neighbours", position);
                return;
            }
            else if (CompareNeighbours(array, position))
            {
                Console.WriteLine("Element at position {0} is larger than neighbours {1} and {2}", position, array[position - 1], array[position + 1]);
            }
            else
            {
                Console.WriteLine("Element at position {0} is not larger than its neighbours", position);
            }
        }
    }
}

File: JavaScript-Fundamentals/Functions/06.LargerThanNeighbours/LargerThanNeighbours.js
// Write a function that checks if the element at given position in given array of integers is bigger than its two neighbours (when such exist).

var jsConsole;
var array = [5, 13, 34, 20, 4 ,10, 45, 30];
var position = 2;

function compareNeighbours (array, position) {
	var isLarger = false;
	if (position < array.length && position > 0) {
		if (array[position] > array[position-1] && array[position] > array[position+1]){
			isLarger = true;
		}
	}
	return isLarger;
}

jsConsole.writeLine('We have array with elements:' + array);
jsConsole.writeLine('The test position is:' + position);
jsConsole.writeLine('Is element at positon ' + position +' larger than its neighbours? ' + (compareNeighbours(array,position) ? 'Yes' : 'No'));